src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java

```diff
@@ -849,25 +852,45 @@ public void invoke() {
 	}
 	
 	public void onDeleteWiki() {
-		// Confirm the delete with the user.
-		view.showConfirmDialog(CONFIRM_DELETE_TITLE,ARE_YOU_SURE_YOU_WANT_TO_DELETE+" this wiki page and all subpages?", new Callback() {
+		WikiPageKey key = new WikiPageKey(this.entityBundle.getEntity().getId(), ObjectType.ENTITY.name(), wikiPageId);
+		// Get the wiki page title and parent wiki id.  Go to the parent wiki if this delete is successful.
+		synapseClient.getV2WikiPage(key, new AsyncCallback<V2WikiPage>() {
 			@Override
-			public void invoke() {
-				postConfirmedDeleteWiki();
+			public void onSuccess(V2WikiPage page) {
+				// Confirm the delete with the user.
+				final String parentWikiId = page.getParentWikiId();
+				String confirmMessage;
+				if (parentWikiId == null) {
+					confirmMessage = ARE_YOU_SURE_YOU_WANT_TO_DELETE+THE_ROOT_WIKI_PAGE_AND_ALL_SUBPAGES;
+				} else {
+					confirmMessage = ARE_YOU_SURE_YOU_WANT_TO_DELETE+"the wiki page \""+page.getTitle()+"\" and all subpages?";
+				}

```

Comment:
It was only recently that I became aware that this was something that the backend supports.  I've changed it so that it shows the wiki ID in this case (in this confirmation message, and in the wiki subtree)